Added ADO Services PAT documentation (#884)
hashicorp/terraform-docs-common: website/docs/cloud-docs/vcs/azure-devops-services-pat.mdx

---

> page_title: Set up the Azure DevOps Services VCS provider using Personal Access Tokens

Reviewer: > `Personal Access Tokens` is a proper noun; should it not be capitalized?

MS is a little inconsistent, but most of the time they use lower case as in the title of this article: https://learn.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate?view=azure-devops&tabs=Windows.

- **Protect this private key carefully.** It can push code to the repositories you use to manage your infrastructure. Take note of your organization's policies for protecting important credentials and be sure to follow them.

1. On a secure workstation, create an SSH keypair that HCP Terraform can use to connect to Azure DevOps Services.com. The exact command depends on your OS, but is usually something like:
   `ssh-keygen -t rsa -m PEM -f "/Users/<NAME>/.ssh/service_terraform" -C "service_terraform_enterprise"`
   This creates a `service_terraform` file with the private key, and a `service_terraform.pub` file with the public key. This SSH key **must have an empty passphrase**. HCP Terraform cannot use SSH keys that require a passphrase.

1. While logged into the Azure DevOps Services account you want HCP Terraform to act as, navigate to the SSH Keys settings page, add a new SSH key and paste the value of the SSH public key you just created.

1. In HCP Terraform's **Add VCS Provider** page, paste the text of the **SSH private key** you just created, and click the **Add SSH Key** button.

> ## Finished
> 
> At this point, Azure DevOps Services access for HCP Terraform is fully configured, and you can create Terraform workspaces based on your organization's repositories.

Reviewer: ```suggestion
## Next steps

Azure DevOps Services access for HCP Terraform is fully configured. You can create Terraform workspaces based on your organization's repositories.
```
